### PR TITLE
Add ref var decl detection for for/foreach

### LINKED ISSUE #37223: Intellisense not listing "var" when starting to write a foreach ref loop
**Version Used**: 
VS16.1.6

**Steps to Reproduce**:

1. Type "foreach" and hit <kbd>tab</kbd> twice
2. Type "ref var" and hit <kbd>tab</kbd>

**Expected Behavior**:

```C#
foreach (ref var item in collection)
{
}
```

**Actual Behavior**:

```C#
foreach (ref EnvironmentVariableTarget item in collection)
{
}
```

### PR DESCRIPTION
**Pull request by diryboy**
Fixes #37223

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/KeywordCompletionProviderTests.cs [hunk 1]
```diff
@@ -414,5 +414,117 @@ public async Task SuggestEventAfterReadonlyInStruct()
...
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(37223, "https://github.com/dotnet/roslyn/issues/37223")]
+        public async Task SuggestVarInForRefLoop0()
+        {
+            var markup =
+@"class C {
+    void F() {
+        for (ref $$
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/KeywordCompletionProviderTests.cs]: teh right place for the tests is in RefKeywordRecommenderTests

that tests these fine-grained scenarios.  this test file is supposed to just be for validating that keywords show up :)

Comment (jasonmalinowski) [reply src/EditorFeatures/CSharpTest/Completion/CompletionProviders/KeywordCompletionProviderTests.cs]: Ah missed that good catch @CyrusNajmabadi.

Comment (diryboy) [reply src/EditorFeatures/CSharpTest/Completion/CompletionProviders/KeywordCompletionProviderTests.cs]: Thanks! I suspected that so few test cases and files here?! It turns out that they're in another project.

BTW, the correct place is `VarKeyword...` not `RefKeyword...`

**Comment by diryboy** [comment]
@jasonmalinowski @CyrusNajmabadi Tests are now moved to RefKeywordRecommenderTests.cs

**Comment by jasonmalinowski** [comment]
Thanks @diryboy for the fix!

**Pull request merged**